feat: Flux package manager that can renovate HelmRelease manifests (#13566)
renovatebot/renovate: lib/manager/flux/readme.md

Checks YAML manifests for [Flux](https://fluxcd.io/) `HelmRelease` resources and extracts dependencies for the `helm` datasource.

> Note that the `flux` manager will only extract `HelmRelease` resources linked to `HelmRepository` sources. `HelmRelease` resources linked to other kinds of sources like `GitRepository` or `Bucket` will be ignored.

Reviewer: ```suggestion
The `flux` manager will only extract `HelmRelease` resources linked to `HelmRepository` sources.
`HelmRelease` resources linked to other kinds of sources like `GitRepository` or `Bucket` will be ignored.
```

We can drop the `Note that` part, and use one sentence per line.

> Also, for the `flux` manager to properly link `HelmRelease` and `HelmRepository` resources, both must have their `namespace` explicitly set in their `metadata`. Dependencies will not be extracted from resources where `metadata.namespace` is omitted. (However, the namespace can be omitted from the `HelmRelease` resource's `spec.chart.spec.sourceRef` property.)

Reviewer: ```suggestion
For the `flux` manager to properly link `HelmRelease` and `HelmRepository` resources, _both_ must have their `namespace` explicitly set in their `metadata`.
Dependencies will not be extracted from resources where `metadata.namespace` is omitted.
However, the namespace can be omitted from the `HelmRelease` resource's `spec.chart.spec.sourceRef` property.
```

Simplify start of sentence, emphasize the _both_ part, use one sentence per line.

> The `flux` manager has no `fileMatch` default patterns, so it won't match any files until you configure it with a pattern. This is because there is no commonly accepted file/directory naming convention for Flux manifests and we don't want to check every single `*.yaml` file in repositories just in case any of them contain Flux definitions.

Reviewer: ```suggestion
The `flux` manager has no `fileMatch` default patterns, so it won't match any files until you configure it with a pattern.
This is because there is no commonly accepted file/directory naming convention for Flux manifests and we don't want to check every single `*.yaml` file in repositories just in case any of them contain Flux definitions.
```

One sentence per line.

Checks YAML manifests for [Flux](https://fluxcd.io/) `HelmRelease` resources and extracts dependencies for the `helm` datasource.

The `flux` manager will only extract `HelmRelease` resources linked to `HelmRepository` sources.

> `HelmRelease` resources linked to other kinds of sources like `GitRepository` or `Bucket` will be ignored.

Reviewer: ```suggestion
`HelmRelease` resources linked to other sources like `GitRepository` or `Bucket` will be ignored.
```

Author: `GitRepository` (and `Bucket`) is a `kind` of Flux source - it's not a source in itself, which is why I phrased it this way.

For the `flux` manager to properly link `HelmRelease` and `HelmRepository` resources, _both_ of the following conditions must be met:

> 1. The `HelmRelease` resource must either have its `metadata.namespace` property set or its `spec.chart.spec.sourceRef.namespace` property set.
> 2. The referenced `HelmRepository` resource must have its `metadata.namespace` property set.

Reviewer: ```suggestion
1. The `HelmRelease` resource must either have its `metadata.namespace` property set or its `spec.chart.spec.sourceRef.namespace` property set
2. The referenced `HelmRepository` resource must have its `metadata.namespace` property set
```

We don't use punctuation at the end of a list.

Namespaces will not be inferred from the context (e.g. from the parent `Kustomization`).

The `flux` manager has no `fileMatch` default patterns, so it won't match any files until you configure it with a pattern.
This is because there is no commonly accepted file/directory naming convention for Flux manifests and we don't want to check every single `*.yaml` file in repositories just in case some of them contain Flux definitions.

If most `.yaml` files in your repository are Flux manifests, then you could add this to your config:

```json
{
  "flux": {
    "fileMatch": ["\\.yaml$"]
  }
}
```

> If instead you have them all inside a `flux/` directory, you would add this:

Reviewer: ```suggestion
If instead you have all your Flux manifests inside a `flux/` directory, you would add this:
```

Explaining what `them` refers makes the sentence easier to understand, when you read the sentence on its own.
